Sources/Payloads/Processing/Helpers/MLSEventProcessor.swift

```diff
@@ -32,40 +32,49 @@ class MLSEventProcessor: MLSEventProcessing {
     // MARK: - Update conversation
 
     func updateConversationIfNeeded(conversation: ZMConversation, groupID: String?, context: NSManagedObjectContext) {
+        Logging.mls.info("MLS event processor updating conversation if needed")
+
         guard conversation.messageProtocol == .mls else {
-            return Logging.mls.info("Message protocol is not mls")
+            return Logging.mls.warn("MLS event processor aborting conversation update: not an MLS conversation")
         }
 
         guard let mlsGroupID = MLSGroupID(from: groupID) else {
-            return Logging.mls.warn("MLS group ID is missing or invalid")
+            return Logging.mls.warn("MLS event processor aborting conversation update: missing group ID")
         }
 
+        let previousGroupID = conversation.mlsGroupID
         conversation.mlsGroupID = mlsGroupID
 
+        Logging.mls.info("MLS event processor updated previous group ID (\(String(describing: previousGroupID))) with new value (\(mlsGroupID))")
```

Comment:
This might be clearer, what do you think?

```suggestion
        if conversation.mlsGroupID == nil {
           conversation.mlsGroupID = mlsGroupID
           Logging.mls.info("MLS event processor set the group ID to value: (\(mlsGroupID)) for conversation: (\(String(describing: conversation.qualifiedID))")
        }
```